src/sgb-v/message.ts

```diff
@@ -0,0 +1,141 @@
+/** based on documents: 
+ *  - Sonstige Leistungserbringer, Technische Anlage 1 für die maschinelle Abrechnung
+ * 
+  * see docs/documents.md for more info
+  */
+
+import { 
+    TestIndicator
+} from "../types"
+import { 
+    getSummenstatus,
+    LeistungserbringerSammelgruppenSchluessel,
+    SummenstatusSchluessel,
+    summenstatusSchluessel,
+} from "./codes"
+import { elements } from "../edifact/builder"
+import { date, time, char, fixedInt } from "../edifact/formatter"
+import { Message, Segment } from "../edifact/types"
+import { 
+    FKT, 
+    FKT_Sammelrechnung, 
+    GES, 
+    NAM, 
+    REC, 
+    REC_Sammelrechnung, 
+    SKO, 
+    UST
+} from "./segments_slga"
+import { 
+    Einzelrechnung, 
+    Sammelrechnung, 
+    Skonto,
+    BaseAbrechnungsfall,
+    Gesamtsummen
+} from "./types"
+
+export const makeInterchangeHeader = (
+    /** IK of the sender (creator) of this bill */
+    absenderIK: string,
+    /** IK of the designated recipient of this file */
+    encryptedForIK: string,
+    /** date at which this file has been created */
+    dateCreated: Date,
+    /** serial number that should be increased by one for each transmission to the recipient
+     *  (encryptedForIK). A value from 00001-99999. It should loop back to 00001 before 100000 is 
+     *  reached. */
+    datenaustauschreferenz: number,
+    /** To which group of health care providers the care provider belongs */
+    leistungsbereich: LeistungserbringerSammelgruppenSchluessel,
+    /** AKA "logischer Dateneiname" */
+    anwendungsreferenz: string,
+    /** Whether this bill is just a test or its real data */
+    testIndicator: TestIndicator
+) => elements(
+    ["UNOC", "3"],
+    char(absenderIK, 9),
+    char(encryptedForIK, 9),
+    [date(dateCreated), time(dateCreated)],
+    fixedInt(datenaustauschreferenz, 5),
+    leistungsbereich,
+    char(anwendungsreferenz, 11),
+    testIndicator
+)
+
+export const makeSLGA_SammelrechnungMessage = <T extends BaseAbrechnungsfall>(
+    s: Sammelrechnung,
+    abrechnungsfaelle: T[],
+    calculateGesamtsummen: (abrechnungsfaelle: T[]) => Gesamtsummen
+): Message => ({
+    header: elements(["SLGA", "16", "0", "0"]),
+    segments: [
+        /* NOTE: if any other verarbeitungskennzeichen than "01" is supported, the calculation of 
+                GES needs to be adjusted */ 
```

Comment:
I don't think we need to deal with VKZ other than 01. In SGB-V, if I remember correctly, anything else than 01 is only supported for one Leistungerbringersammelgruppe and only since half a year or so: For Heilmittelerbringer. So, not relevant for us.

Reply:
Good to know! Let's add that to the comment.
```suggestion
        /* NOTE: if any other verarbeitungskennzeichen than "01" is supported, the calculation of 
                GES needs to be adjusted, which only applies to Heilmittelerbringer right now */ 
```